MoreLinq/HasDuplicates.cs

```diff
@@ -0,0 +1,56 @@
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        ///   Checks if sequence contains duplicates
+        /// </summary>
+        /// <param name="source">The sequence to check.</param>
+        /// <typeparam name="T">The type of the elements in the source sequence</typeparam>
+        /// <returns>true if any element of the sequence is duplicated, false otherwise</returns>
+        public static bool HasDuplicates<T>(this IEnumerable<T> source) => source?.HasDuplicates(EqualityComparer<T>.Default) ?? throw new ArgumentNullException(nameof(source));
+
+        /// <summary>
+        ///   Checks if sequence contains duplicates, using the specified element equality comparer
+        /// </summary>
+        /// <param name="source">The sequence to check.</param>
+        /// <param name="keySelector">Projection for determining "distinctness"</param>
+        /// <typeparam name="TSource">Type of the source sequence</typeparam>
+        /// <typeparam name="TKey">Type of the projected element</typeparam>
+        /// <returns>true if any element of the sequence is duplicated, false otherwise</returns>
+        public static bool HasDuplicates<TSource, TKey>(this IEnumerable<TSource> source, Func<TSource, TKey> keySelector) => source?.HasDuplicates(keySelector, EqualityComparer<TKey>.Default) ?? throw new ArgumentNullException(nameof(source));
+
+        /// <summary>
+        ///   Checks if sequence contains duplicates, using the specified element equality comparer
+        /// </summary>
+        /// <param name="source">The sequence to check.</param>
+        /// <param name="comparer">The equality comparer to use to determine whether or not keys are equal.
+        /// If null, the default equality comparer for <c>TSource</c> is used.</param>
+        /// <typeparam name="T">The type of the elements in the source sequence</typeparam>
+        /// <returns>true if any element of the sequence is duplicated, false otherwise</returns>
+        public static bool HasDuplicates<T>(this IEnumerable<T> source, IEqualityComparer<T>? comparer) => source?.HasDuplicates(x => x, comparer) ?? throw new ArgumentNullException(nameof(source));
```

Comment:
Should use `IdFn` instead of `x => x`; for JIT/perf/etc. reasons.

Reply:
nice, didn't know about that! I'll swap it thanks!